src/app/operations/operations-db.service.ts

```diff
@@ -92,6 +107,17 @@ export class OperationDbService {
     return operations.filter((operation: Operation) => operation.id === id)[0];
   }
 
+  toggleDoneState(operation: Operation) {
+    console.log(operation.id);
+    this.storage.get('operation').then((operations) => {
+      const currOp = operations.filter((operat: Operation) => operat.id === operation.id)[0];
+      if (currOp.id === operation.id) {
```

Comment:
The if statement is unnecessary : the id will always be equal, because that's what is tested in the `filter`.
Also, I would rename `currOp` to `storedOp` and `operat` to `currOp` to make it a bit more understandable for others